### PR TITLE
fix: PostMessage's message field in Action selector doesn't support Objects

### LINKED ISSUE #23167: [Bug]-[544]:Post message issues
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Add message for Action selector - postmessage -

1. Try typing bellow code in message box - 
`{{{ a:1, b:API1.data.users[0].name}}}`
Actual - only {{{}}} bracket is saved

2. Try adding below code 
'{{{a:1,b:2}}}'
Actual - only {a:1,b:2} is saved

https://github.com/appsmithorg/appsmith/assets/80756091/61b25b5a-9741-4aaf-b047-069404f0f85b

Message box is not storing the data properly

https://github.com/appsmithorg/appsmith/assets/80756091/80637eec-a851-48e4-9242-c5dcc40f7c89

### Steps To Reproduce

Post-message - message box is not storing the data properly

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

cloud

### PR DESCRIPTION
**Pull request by arunvjn**
#### PR fixes following issue(s)
Fixes #23167
>
>
#### Type of change
- Bug fix (non-breaking change which fixes an issue)
>
>
>
#### How Has This Been Tested?
- [x] Manual
- [ ] Jest
- [x] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by Rishabh-Rathod — APPROVED** [review]
LGTM

File: app/client/cypress/e2e/Regression/ClientSide/ActionExecution/ActionSelector_JsToNonJSMode_2_spec.ts [hunk 1]
```diff
@@ -646,4 +646,43 @@ describe("JS to non-JS mode in Action Selector", () => {
       1,
     );
   });
+
+  it("10. Message field in PostMessage should accept all type of values", () => {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/ActionExecution/ActionSelector_JsToNonJSMode_2_spec.ts:650]: Add bug id to `it` description & add the test case to separate spec - ActionSelector_JsToNonJSMode_4_spec. The spec is already too big for any failure analysis & we attempted to split it.

**Review by Aishwarya-U-R — COMMENTED** [review]
Except review comment nit, Cypress looks ok!

File: app/client/cypress/e2e/Regression/ClientSide/ActionExecution/ActionSelector_JsToNonJSMode_2_spec.ts (hunk 1 above)
Comment (arunvjn) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/ActionSelector_JsToNonJSMode_2_spec.ts:650]: Will add the bug ID but was the test limit 10 per file ?

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/ActionSelector_JsToNonJSMode_2_spec.ts]: Yes 10 tests, but if lines are code are more than 300, we will need to split irrespective of the test case count reached or not.

**Review by Aishwarya-U-R — COMMENTED** [review]
Approving Cypress changes.

File: app/client/cypress/e2e/Regression/ClientSide/ActionExecution/ActionSelector_JsToNonJSMode_2_spec.ts (hunk 1 above)
Comment (arunvjn) [reply app/client/cypress/e2e/Regression/ClientSide/ActionExecution/ActionSelector_JsToNonJSMode_2_spec.ts]: Moved the test to a new file

**Comment by shwetha-ramesh** [comment]
@arunvjn tested on [DP](https://ce-25467.dp.appsmith.com/). Working as expected. Moving to done.

**Comment by Rishabh-Rathod** [comment]
> @arunvjn tested on [DP](https://ce-25467.dp.appsmith.com/). Working as expected. Moving to done.

Added test plan approved label according to above comment

**Pull request merged**